helpers/inject.js

```diff
@@ -1,6 +1,37 @@
 'use strict';
 
 const factory = globals => {
+    function filterValues(value) {
+        let result = value;
+        try {
+            JSON.parse(value);
+        } catch (e) {
+            if (typeof value === 'string') {
+                result = globals.handlebars.escapeExpression(value);
+            }
+            if (typeof value === 'object' && value !== null) {
+                result = filterObjectValues(value);
+            }
+        }
+        return result;
+    }
+    function filterObjectValues(obj) {
+        let filteredObject = {};
+        Object.keys(obj).forEach(key => {
+            if (typeof obj[key] === 'string') {
+                filteredObject[key] = globals.handlebars.escapeExpression(obj[key]);
+                return;
+            }
+            if (typeof obj[key] === 'object' && obj[key] !== null) {
+                filteredObject[key] = filterObjectValues(obj[key]);
+                return;
+            } else {
+                filteredObject[key] = obj[key];
```

Comment:
should there be an else case ? and should we be adding the data as filtered without verifying it ?

Reply:
The else might not be necessary, since there's a return on the previous 2 conditions, so I could just assign the value to the filtered object.

I'm not sure what else I'd be verifying, since other types should be fine to include without worrying about XSS. The purpose of the filterObjectValues function is to make sure any strings that can be injected are escaped, even if they are in nested objects/arrays.